packages/evm/test/precompiles/05-modexp.spec.ts

```diff
@@ -51,3 +80,44 @@ describe('Precompiles: MODEXP', () => {
     assert.strictEqual(result.executionGasUsed, gas)
   })
 })
+
+describe('Precompiles: MODEXP with EIP-7823', async () => {
+  const common = new Common({ chain: Mainnet, eips: [7823] })
+  const evm = await createEVM({
+    common,
+  })
+  const addressStr = '0000000000000000000000000000000000000005'
+  const MODEXP = getActivePrecompiles(common).get(addressStr)!
+
+  for (let i = 1020; i < 1028; i++) {
+    const bLen = i
+    const eLen = i
+    const mLen = i
+    const maxInputLen = Math.max(bLen, eLen, mLen)
+    const base = randomBytes(bLen)
+    const exponent = randomBytes(eLen)
+    const modulo = randomBytes(mLen)
+    const expected = modPow(bytesToBigInt(base), bytesToBigInt(exponent), bytesToBigInt(modulo))
+    const input = concatBytes(
+      setLengthLeft(toBytes(bLen), 32),
+      setLengthLeft(toBytes(eLen), 32),
+      setLengthLeft(toBytes(mLen), 32),
+      concatBytes(base, exponent, modulo),
+    )
+
+    it(`MODEXP should reject and consume all gas for inputs over 8192 bits in length - case ${maxInputLen} bytes`, async () => {
+      const result = await MODEXP({
+        data: input,
+        gasLimit: BigInt(0xffffffff),
```

Comment:
I think there should be a sanity check to confirm that this gasLimit is enough to pay for the precompile,  but that the precompile now rejects it due to the incode length limit